mapper/orm/src/main/java/org/hibernate/search/mapper/orm/outbox/impl/OutboxTableIndexerExecutor.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.orm.outbox.impl;
+
+import static org.hibernate.search.mapper.orm.common.spi.TransactionUtils.withinTransaction;
+
+import java.lang.invoke.MethodHandles;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+import org.hibernate.search.mapper.orm.automaticindexing.spi.AutomaticIndexingMappingContext;
+import org.hibernate.search.mapper.orm.automaticindexing.spi.AutomaticIndexingQueueEventProcessingPlan;
+import org.hibernate.search.mapper.orm.logging.impl.Log;
+import org.hibernate.search.mapper.pojo.route.DocumentRoutesDescriptor;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+import org.hibernate.search.util.common.serialization.spi.SerializationUtils;
+
+public class OutboxTableIndexerExecutor {
+
+	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
+	private static final int MAX_BATCH_SIZE = 50;
+
+	private final AutomaticIndexingMappingContext mapping;
+	private final Set<Integer> ids = new HashSet<>();
+	private final ScheduledExecutorService executor;
+
+	public OutboxTableIndexerExecutor(AutomaticIndexingMappingContext mapping,
+			ScheduledExecutorService executor) {
+		this.mapping = mapping;
+		this.executor = executor;
+	}
+
+	public void start() {
+		executor.scheduleAtFixedRate( this::run, 0, 8, TimeUnit.MILLISECONDS );
+	}
+
+	public CompletableFuture<?> stop() {
+		executor.shutdown();
```

Comment:
this call is not to dispose the executor resource, but to return something similar to `BatchingExecutor#completition`.

Reply:
The point of returning a `CompletableFuture<?>` is to be non-blocking, so this is not ideal.
Anyway, using `SingletonTask` as I suggested above should make this problem irrelevant, since `SingletonTask` can give you a `CompletableFuture<?>` through its `completion()` method.

Reply:
right! But I just said that to explain the reason why I did it.